src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_QuantumForceTransformer.java

```diff
@@ -0,0 +1,1160 @@
+                return -1;
+            } else if (block == ModBlocks.blockCasings5Misc) { // Glass
+                return 1;
+            }
+            return -1;
+        };
+    }
+
+    private void setChamberTier(int tier) {
+        mChamberTier = tier;
+    }
+
+    private void setFabCoilTier(int tier) {
+        mFabCoilTier = tier;
+    }
+
+    private void setGlassTier(int tier) {
+        mGlassTier = tier;
+    }
+
+    private int getChamberTier() {
+        return mChamberTier;
+    }
+
+    private int getFabCoilTier() {
+        return mFabCoilTier;
+    }
+
+    private int getGlassTier() {
+        return mGlassTier;
+    }
+
+    @Override
+    public String getSound() {
+        return GregTech_API.sSoundList.get(Integer.valueOf(208));
+    }
+
+    @Override
+    protected IIconContainer getActiveOverlay() {
+        return TexturesGtBlock.Overlay_Machine_Controller_Advanced_Active;
+    }
+
+    @Override
+    protected IIconContainer getInactiveOverlay() {
+        return TexturesGtBlock.Overlay_Machine_Controller_Advanced;
+    }
+
+    @Override
+    protected int getCasingTextureId() {
+        return 70;
+    }
+
+    @Override
+    public boolean hasSlotInGUI() {
+        return true;
+    }
+
+    @Override
+    public String getCustomGUIResourceName() {
+        return null;
+    }
+
+    @Override
+    public GT_Recipe.GT_Recipe_Map getRecipeMap() {
+        return GTPP_Recipe.GTPP_Recipe_Map.sQuantumForceTransformerRecipes;
+    }
+
+    @Override
+    public boolean isCorrectMachinePart(final ItemStack aStack) {
+        return true;
+    }
+
+    @Override
+    public long getMaxInputVoltage() {
+        long rVoltage = 0;
+        for (GT_MetaTileEntity_Hatch_Energy tHatch : mEnergyHatches)
+            if (isValidMetaTileEntity(tHatch)) {
+                rVoltage += tHatch.getBaseMetaTileEntity().getInputVoltage()
+                        * tHatch.getBaseMetaTileEntity().getInputAmperage();
+            }
+        return rVoltage;
+    }
+
+    private int mMaxParallel = 64;
+    private int mCurrentParallel = 0;
+
+    @Override
+    public boolean checkRecipe(final ItemStack aStack) {
+        int mCurrentParallel = 0;
+        this.mEUt = 0;
+        this.mMaxProgresstime = 0;
+        this.mOutputItems = null;
+        this.mOutputFluids = null;
+        FluidStack[] tFluidList = getCompactedFluids();
+        ItemStack[] tCatalysts = mCatalystBuses.get(0).mInventory;
+        if (mSeparateInputBusses) {
+            ArrayList<ItemStack> tInputList = new ArrayList<ItemStack>();
+            for (GT_MetaTileEntity_Hatch_InputBus tBus : mInputBusses) {
+                for (int i = tBus.getSizeInventory() - 1; i >= 0; i--) {
+                    if (tBus.getStackInSlot(i) != null) {
+                        tInputList.add(tBus.getStackInSlot(i));
+                        if (ItemUtils.isCatalyst(tBus.getStackInSlot(i))) {
+                            if (mCurrentParallel < mMaxParallel) {
```

Comment:
Why is parallel logic here when its also in `processRecipe`, which is also called within this if-condition?

Reply:
I forgot to remove it. Old things from my first iteration.